packages/flutter/lib/src/material/back_button.dart

```diff
@@ -84,7 +87,10 @@ class BackButton extends StatelessWidget {
       color: color,
       tooltip: 'Back', // TODO(ianh): Figure out how to localize this string
       onPressed: () {
-        Navigator.of(context).maybePop();
+        Navigator.of(context).maybePop().then((bool didPop) {
+          if (!didPop)
```

Comment:
This isn't where this should be implemented. We should have a central location where anyone hitting back, whether via the Android system back button, or this button, or just arbitrarily calling pop(), or whatever, gets the same result. MaterialApp's logic, maybe.